### PR TITLE
Add support for Apicurio Registry 2.x Avro library

### PR DESCRIPTION
**Pull request by Ladicek**

### CODE REVIEW

**Comment by Ladicek** [comment]
Draft because Apicurio Registry 2.x hasn't been released yet.

File: extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java [hunk 1]
```diff
@@ -103,6 +106,12 @@
 
     static final DotName OBJECT_MAPPER = DotName.createSimple("com.fasterxml.jackson.databind.ObjectMapper");
 
+    // TODO do we want this?
```
Comment (gsmet) [inline extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java:109]: Yes we do and we can remove this comment :).

**Comment by Ladicek** [comment]
And also I need to add a test :-)

**Review by gsmet — DISMISSED** [review]
Looks good. I added a small comment, please dismiss when fixed

File: extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java (hunk 1 above)
Comment (Ladicek) [reply extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java:109]: Thanks! :-)

**Review by michalszynkiewicz — APPROVED** [review]

**Comment by Ladicek** [comment]
To fix the test failure, I rebased my branch and added https://github.com/quarkusio/quarkus/pull/16251/files#diff-6d16e1bd2cec46a1c2d8f00ec4f3476b7554df6379ac769a0492f8aad9f69911 to the new `kafka-avro-apicurio2` test suite module.

**Comment by Ladicek** [comment]
That looks quite unrelated.

**Comment by Ladicek** [comment]
Sorry, I don't know how to properly "dismiss" the request for change :-)

**Comment by Ladicek** [comment]
Ha, now I know how to dismiss the request :-) Haha, sorry for the noise!

I also rebased and re-pushed, maybe the unrelated flaky test will pass this time.

**Comment by Ladicek** [comment]
Rebased, as #16746 might have fixed the MySQL issue.

**Review by cescoffier — APPROVED** [review]
Awesome!

**Pull request merged**